src/__tests__/element-queries.js

```diff
@@ -801,4 +801,70 @@ test('get/query textarea element by current value', () => {
   expect(queryByDisplayValue('World').id).toEqual('content-textarea')
 })
 
+test('DescriptionOf uses aria-describedby', () => {
+  const {
+    getByRole,
+    getDescriptionOf,
+    queryDescriptionOf,
+    findDescriptionOf,
+  } = renderIntoDocument(`
+    <main>
+      <button aria-describedby="button-description">Click me</button>
+      <p id="button-description">Hello, Dave!</p>
+    </main>
+  `)
+
+  expect(getDescriptionOf(getByRole('button'))).toHaveTextContent(
+    'Hello, Dave!',
+  )
+  expect(queryDescriptionOf(getByRole('button'))).toHaveTextContent(
+    'Hello, Dave!',
+  )
+  expect(findDescriptionOf(getByRole('button'))).resolves.toHaveTextContent(
+    'Hello, Dave!',
+  )
+})
+
+test('AllDescriptionOf throws with a recommendation', () => {
+  const {
+    getByRole,
+    getDescriptionOf,
+    getAllDescriptionOf,
+    queryAllDescriptionOf,
+    findAllDescriptionOf,
+  } = renderIntoDocument(`
+    <main>
+      <button aria-describedby="button-description">Click me</button>
+      <p id="button-description">Hello, Jekyll!</p>
+      <p id="button-description">Hello, Hyde!</p>
+    </main>
+  `)
+
+  expect(() => getDescriptionOf(getByRole('button'))).toThrow(
+    'Found multiple descriptions. An element should be described by a unique element.',
+  )
+  expect(() => getAllDescriptionOf(getByRole('button'))).toThrow(
+    'An element should be described by a unique element. If you want to expect that only a single element exists prefer getDescriptionOf and expect an error thrown.',
+  )
+  expect(() => queryAllDescriptionOf(getByRole('button'))).toThrow(
+    'An element should be described by a unique element. If you want to expect that only a single element exists prefer queryDescriptionOf and expect an error thrown.',
+  )
+  expect(() => findAllDescriptionOf(getByRole('button'))).toThrow(
```

Comment:
Same here

Reply:
It doesn't return a promise but immediately throws. Should it return a promise that rejects instead?

Reply:
Sorry. Was confused. This makes sense.